specification/orbital/resource-manager/Microsoft.Orbital/stable/2022-11-01/orbital.json

```diff
@@ -0,0 +1,2219 @@
+        "Communication"
+      ],
+      "x-ms-enum": {
+        "name": "Capability",
+        "modelAsString": true
+      }
+    },
+    "ResourceIdListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "id": {
+                "type": "string",
+                "description": "The Azure Resource ID."
+              }
+            }
+          },
+          "description": "A list of Azure Resource IDs."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Response for an API service call that lists the resource IDs of resources associated with another resource."
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "enum": [
+        "creating",
+        "succeeded",
+        "failed",
+        "canceled",
+        "updating",
+        "deleting"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      },
+      "description": "The current state of the resource's creation, deletion, or modification.",
+      "readOnly": true
+    },
+    "AuthorizedGroundstation": {
+      "type": "object",
+      "required": [
+        "groundStation",
+        "expirationDate"
+      ],
+      "readOnly": true,
+      "properties": {
+        "groundStation": {
+          "type": "string",
+          "description": "Groundstation name."
+        },
+        "expirationDate": {
+          "type": "string",
+          "format": "date",
+          "description": "Date of authorization expiration."
+        }
+      },
+      "description": "Authorized groundstation."
+    },
+    "OperationResult": {
+      "description": "Operation Result Entity.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "description": "ID of the resource.",
+          "type": "string"
+        },
+        "name": {
+          "readOnly": true,
+          "description": "Name of the resource.",
+          "type": "string"
+        },
+        "status": {
+          "$ref": "#/definitions/Status"
+        },
+        "startTime": {
+          "description": "The operation start time (ISO 8601 UTC standard).",
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time"
+        },
+        "endTime": {
+          "description": "The operation end time (ISO 8601 UTC standard).",
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time"
+        },
+        "percentComplete": {
+          "type": "number",
+          "format": "double",
+          "minimum": 0,
+          "maximum": 100,
+          "description": "Percentage completed.",
+          "readOnly": true
+        },
+        "value": {
```

Comment:
>value [](http://example.com/codeflow?start=9&length=5)

I'm quite confused about why your OperationResult payload includes these.

Are you just trying to reuse the same class for LIST of operationResults and GET single operationResult? Or do you actually return both sets of information in a single response?

Reply:
Hey Tim, you're referring to the "value" and "nextLink" fields right? I understand the confusion, and it's not for the reason you've mentioned.

Right now we have several long-running operations within our API, such as PUT Contact, PUT Contact Profile, etc. Those all return their result through operationResult in the way you'd expect.

One of our long-running APIs though is POST spacecrafts/contacts/listAvailableContacts; this one returns a _list_ of visibility windows with a customer's spacecraft rather than returning a single result. Being long-running, it returns its result through the operationResult as well, and its response looks like the standard paging response body for list results. 

We originally had this as a synchronous API, and did not want to change the response body. As a result, we needed to add nextLink and value as optional fields to operationResult, because the POST spacecrafts/contacts/listAvailableContacts API has that format.

Let us know if you have any thoughts on this since I know it looks a bit wonky.

Thanks! We appreciate you being so thorough.

Reply:
So... most async POST APIs using the 202 Accepted pattern and Location header **shouldn't actually return a body**. If you want to actually return a body, then I think we should ask @pilor about the recommended way to do that.

Reply:
Are you positive about that? The Async API Reference in the ARM RPC says nothing of the sort: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers, and it says "If the operation is complete, return the exact same response that would have been returned had the operation been executed synchronously," which is what we're doing here.

In fact, it lists response body as one of the things we should return once the operation is completed: "If the POST completes successfully, the URL that was returned in the Location header MUST now return what would have been a successful response if the API completed (e.g. a response body / header / status code). It is acceptable for this to be a 200/204 NoContent if the action does not require a response (e.g. restarting a VM)."

The only mention I'm seeing of not returning a response body is that the URI returned by the Location header should return a 202 with an empty response body while the operation is _still in progress_.

If we're required to implement this in some different way, that'd be quite the curve ball.

What special concerns are you alluding to? RBAC/Authz concerns, or something else?

Thanks!

Reply:
It is an issue thats okay to address this in a future API version. #resolved

Reply:
Roger. @TimLovellSmith or @pilor , is there anywhere I can read about some more details about the concerns related to these APIs? Would like to plan for any necessary revisions.

Thanks!